doctr/datasets/iiit5k.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2021, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+import os
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import numpy as np
+import scipy.io as sio
+
+from .datasets import VisionDataset
+
+__all__ = ['IIIT5K']
+
+
+class IIIT5K(VisionDataset):
+    """IIIT-5K dataset from `"BMVC 2012 Scene Text Recognition using Higher Order Language Priors"
+    <https://cdn.iiit.ac.in/cdn/cvit.iiit.ac.in/images/Projects/SceneTextUnderstanding/home/mishraBMVC12.pdf>`_.
+
+    Example::
+        >>> from doctr.datasets import IIIT5K
+        >>> train_set = IIIT5K(train=True, extract_archive=True, download=True)
+        >>> img, target = train_set[0]
+
+    Args:
+        train: whether the subset should be the training one
+        sample_transforms: composable transformations that will be applied to each image
+        **kwargs: keyword arguments from `VisionDataset`.
+    """
+
+    DATA_URL = 'https://cvit.iiit.ac.in/images/Projects/SceneTextUnderstanding/IIIT5K-Word_V3.0.tar.gz'
+
```

Comment:
let's add the sha256 as well, it's:
```
7872c9efbec457eb23f3368855e7738f72ce10927f52a382deb4966ca0ffa38e
```

And just so you know, you can compute this for a given file with:
```
sha256sum /path/to/file
```
;)